charms/kfp-ui/src/charm.py

```diff
@@ -88,11 +89,10 @@ def _main(self, event):
             "KUBEFLOW_USERID_PREFIX": "",
             "METADATA_ENVOY_SERVICE_SERVICE_HOST": "localhost",
             "METADATA_ENVOY_SERVICE_SERVICE_PORT": "9090",
-            "MINIO_ACCESS_KEY": os["access-key"],
+            "minio-secret": {"secret": {"name": "minio-secret"}},
```

Comment:
Re the name of the secret, with this name the secret will conflict with the secret our minio charm creates for itself.  This should be named something like `CHARM_NAME-credentials`

Also, by statically defining it (eg: `kfp-ui-credentials`) rather than dynamically (eg: `"{CHARM_NAME}-credentials"`) we also have a problem if we deploy the same charm twice in one model.  That probably isn't an issue here, but it could be.